lib/active_merchant/billing/gateways/realex.rb

```diff
@@ -145,7 +145,11 @@ def build_purchase_or_authorization_request(action, money, credit_card, options)
           add_card(xml, credit_card)
           xml.tag! 'autosettle', 'flag' => auto_settle_flag(action)
           add_signed_digest(xml, timestamp, @options[:login], sanitize_order_id(options[:order_id]), amount(money), (options[:currency] || currency(money)), credit_card.number)
-          add_network_tokenization_card(xml, credit_card) if credit_card.is_a?(NetworkTokenizationCreditCard)
+          if credit_card.is_a?(NetworkTokenizationCreditCard)
+            add_network_tokenization_card(xml, credit_card)
+          else
```

Comment:
Do we need this `else`? We check `if options[:three_d_secure]` in `add_three_d_secure` and there may be cases were we'll want to pass this option for a network tokenization card.

Reply:
I think yes, there's already some logic for cavv and eci, but it's pulled from payment for tokenized cards: https://github.com/activemerchant/active_merchant/blob/8535adc31d6e4285924b8418b61699aee1424fe7/lib/active_merchant/billing/gateways/realex.rb#L279